### PR TITLE
New Data Source: `azurerm_virtual_desktop_application_group`

### LINKED ISSUE #13811: Support for azurerm_virtual_desktop_* data sources.
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It is not currently possible to reference an `azurerm_virtual_desktop_*` resource as a data source. This prevents someone from, for example, creating diagnostic monitoring for an Azure Virtual Desktop Workspace using Terraform. 

If you attempt to reference a azurerm_virtual_desktop_* resource as a data source, this naturally results in an error:

```
Invalid data source
  on data_avd_workspaces.tf line 1, in data "azurerm_virtual_desktop_workspace" "example_workspace":
   1: data "azurerm_virtual_desktop_workspace" "example_workspace" {

The provider provider.azurerm.example_subscription does not support data
source "azurerm_virtual_desktop_workspace".
```

### New or Affected Resource(s)

data "azurerm_virtual_desktop_application"
data "azurerm_virtual_desktop_application_group"
data "azurerm_virtual_desktop_host_pool"
data "azurerm_virtual_desktop_workspace"

### Potential Terraform Configuration

```hcl
data "azurerm_virtual_desktop_application" "example_application" {
  name                = "example_application"
  resource_group_name = "example_resource_group"
}

data "azurerm_virtual_desktop_application_group" "example_application_group" {
  name                = "example_application_group"
  resource_group_name = "example_resource_group"
}

data "azurerm_virtual_desktop_host_pool" "example_host_pool" {
  name                = "example_host_pool"
  resource_group_name = "example_resource_group"
}

data "azurerm_virtual_desktop_workspace" "example_workspace" {
  name                = "example_workspace"
  resource_group_name = "example_resource_group"
}
```

Expected properties to be exposed by these data sources would minimally include the resource's name and ID. 

### References

N/A

### LINKED ISSUE #24625: Support for data block for azurerm_virtual_desktop_application_group
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

As per this [#14146](https://github.com/hashicorp/terraform-provider-azurerm/issues/14146), I am unable to make refernce to existing DAG using data block.
Can this be made avilable please? 

### New or Affected Resource(s)/Data Source(s)

azurerm_virtual_desktop_application_group

### Potential Terraform Configuration

```hcl
data "azurerm_virtual_desktop_application_group" "dag" {
    name = "host-pool-DAG"
    rescource_group_name = "avd-test"
} 

resource "azurerm_role_assignment" "desktop-virtualisation-user" {
  scope = data.azurerm_virtual_desktop_application_group.dag.id
  role_definition_name = "Desktop Virtualization User"
  principal_id         = "xxxxxxxxxx"
}
```

Error:

```
╷
│ Error: Invalid data source
│
│   on test.tf line 1, in data "azurerm_virtual_desktop_application_group" "dag":
│    1: data "azurerm_virtual_desktop_application_group" "dag" {
│
│ The provider hashicorp/azurerm does not support data source "azurerm_virtual_desktop_application_group".
│
│ Did you intend to use the managed resource type "azurerm_virtual_desktop_application_group"? If so, declare this using a "resource" block
│ instead of a "data" block.
```

### References

https://github.com/hashicorp/terraform-provider-azurerm/issues/14146

### PR DESCRIPTION
**Pull request by ASHR4**
# Description

Limited support for desktop virtualization data sources

This PR is creating a new data source for virtual desktops application groups 

# Acceptance Tests

image

# Fixes

Resolves #13811 & Resolves #24625

### CODE REVIEW

File: internal/services/desktopvirtualization/validate/application_group_name.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	// The value must not be empty.
+	if strings.TrimSpace(v) == "" {
+		errors = append(errors, fmt.Errorf("%q must not be empty", k))
+		return
+	}
+
+	const minLength = 3
+	const maxLength = 64
+
+	// Application Group name can be 3-64 characters in length
```
[the remark this answers is not in the document]
Comment (ASHR4) [reply internal/services/desktopvirtualization/validate/application_group_name.go:29]: I've followed the latest documentation which utilises this internal validation.

I've chosen to amend the [application group resource](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/desktopvirtualization/virtual_desktop_application_group_resource.go#L64) to align with this :)

File: internal/services/desktopvirtualization/virtual_desktop_application_group_data_source.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+			state.Location = location.NormalizeNilable(model.Location)
+			state.Tags = pointer.From(model.Tags)
+
+			if model.Properties.ApplicationGroupType != "" && model.Properties.ApplicationGroupType != applicationgroup.ApplicationGroupTypeDesktop {
+				state.ApplicationGroupType = "RemoteApp"
+			} else {
+				state.ApplicationGroupType = "Desktop"
+			}
+
+			if model.Properties.HostPoolArmPath != "" {
```
[the remark this answers is not in the document]
Comment (ASHR4) [reply internal/services/desktopvirtualization/virtual_desktop_application_group_data_source.go]: Nice spot!

**Comment by ASHR4** [comment]
> Hey @ASHR4, this PR is looking good but there are a few differences between the resource implementation and this datasource implementation that we should instead mirror. Let me know what you think!

Thanks @mbfrahry, I'll keep this in mind when contributing in the future! I appreciated the guidance :)

**Review by katbyte — APPROVED** [review]
Thanks @ASHR4 - lGTM ⚖️

**Pull request merged**